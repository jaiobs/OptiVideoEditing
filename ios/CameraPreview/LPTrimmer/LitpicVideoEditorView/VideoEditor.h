//
//  VideoEditor.h
//  react-native-litpic-camera-module
//
//  Created by optisol on 02/08/20.
//

#import <UIKit/UIKit.h>
#import <AVFoundation/AVFoundation.h>
#import <React/RCTViewManager.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

NS_ASSUME_NONNULL_BEGIN
@class VideoEditor;

@interface VideoEditor : RCTViewManager <RCTBridgeModule>

@end

NS_ASSUME_NONNULL_END
