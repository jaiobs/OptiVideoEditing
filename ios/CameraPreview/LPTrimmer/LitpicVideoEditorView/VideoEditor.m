//
//  VideoEditor.m
//  react-native-litpic-camera-module
//
//  Created by optisol on 02/08/20.
//

#import "VideoEditor.h"
#import "RNFileSystem.h"
#import "RNImageUtils.h"
#import <React/RCTBridge.h>
#import <React/RCTUIManager.h>
#import <React/RCTEventDispatcher.h>
#import <React/RCTLog.h>
#import <React/RCTUtils.h>
#import <React/UIView+React.h>
#import "react_native_litpic_camera_module-Swift.h"
#import "RNCamera.h"
#import "RNSensorOrientationChecker.h"
#import "CommonEmittor.h"

@implementation VideoEditor

VideoEditorSwift *VEV;

//RNSensorOrientationChecker * sensorOrientationChecker;

RCT_EXPORT_MODULE(VideoEditor);

- (UIView *)view{
    VEV = [[VideoEditorSwift alloc] init];
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(orientationChanged:)
                                                 name:UIApplicationDidChangeStatusBarOrientationNotification
                                               object:nil];
    
    [VEV copyEmitorWithCompletionHandler:^(NSDictionary *response) {
        [self sendData:response];
    }];

    return VEV;
}

- (void)orientationChanged:(NSNotification *)notification{
    UIInterfaceOrientation orientation = [UIApplication sharedApplication].statusBarOrientation;
}

- (void)sendData:(NSDictionary*)response{
    CommonEmittor *events = [[CommonEmittor alloc] init];
    [events setBridge:self.bridge];
    [events sendResponseEvent:response];
}


RCT_CUSTOM_VIEW_PROPERTY(videoUrl, NSString, VideoEditorSwift)
{
    NSLog(@" VIDEO URL %@",[RCTConvert NSString:json]);
    [VEV setVideoUrls:[RCTConvert NSString:json]];
    [VEV layoutSubviews];
}


RCT_CUSTOM_VIEW_PROPERTY(videoUrlArray, NSArray, VideoEditorSwift)
{
    [VEV setVideoUrlsArray:[RCTConvert NSArray:json]];
    [VEV layoutSubviews];
}


RCT_CUSTOM_VIEW_PROPERTY(IsTilt, BOOL, VideoEditorSwift)
{
    [VEV setIsCropperView:[RCTConvert BOOL:json]];
    [VEV layoutSubviews];
}


RCT_EXPORT_METHOD(IsTilt :(BOOL)value){
    [VEV setIsCropperView:value];
    [VEV layoutSubviews];
}

RCT_EXPORT_METHOD(saveVideo:(double)time callback:(RCTResponseSenderBlock)callback){
    dispatch_async(dispatch_get_main_queue(), ^{
        [VEV saveVideoWithCompletionHandler:^(NSDictionary *response) {
            callback(@[response]);
        }];
    });
}

RCT_EXPORT_METHOD(saveVideoWithTextStickerOverlay:(NSArray *)textDict callback:(RCTResponseSenderBlock)callback){
    dispatch_async(dispatch_get_main_queue(), ^{
        [VEV saveVideoWithTextOverlayWithDrawArrayText:textDict completionHandler:^(NSDictionary * response) {
            callback(@[response]);
        }];
    });
}

RCT_EXPORT_METHOD(getCropVideo:(double)time callback:(RCTResponseSenderBlock)callback){
    dispatch_async(dispatch_get_main_queue(), ^{
        [VEV cropVideoDidSelectedWithCompletionHandler:^(NSDictionary *response) {
            callback(@[response]);
        }];
    });
}



RCT_EXPORT_METHOD(getVideoCropFrame:(double)time callback:(RCTResponseSenderBlock)callback){
    dispatch_async(dispatch_get_main_queue(), ^{
        [VEV getVideoCropRectWithCompletionHandler:^(NSDictionary *response) {
                    callback(@[response]);
            }];
    });
}

RCT_EXPORT_METHOD(isMusicSynced:(BOOL)isMusic){
    dispatch_async(dispatch_get_main_queue(), ^{
        [VEV muteSoundOnMusicSyncWithIsflag:isMusic];
    });
}


@end
