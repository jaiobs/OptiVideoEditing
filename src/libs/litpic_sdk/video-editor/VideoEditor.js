import React, {Component} from 'react';
import {
  View,
  requireNativeComponent,
  findNodeHandle,
  UIManager,
  NativeModules,
  NativeEventEmitter,
} from 'react-native';

const VideoEditorComponent = requireNativeComponent('VideoEditorAndroid');
const VideoEditorViewManager = NativeModules.VideoEditorViewModule;

export default class VideoEditor extends Component {
  constructor(props) {
    super(props);
  }

  _setReference = (ref: ?Object) => {
    if (ref) {
      this._videoEditorRef = ref;
      this._videoEditorHandle = findNodeHandle(ref);
    } else {
      this._videoEditorRef = null;
      this._videoEditorHandle = null;
    }
  };

  addTextOverLay() {
    this.executeCommand("add_text_overlay", []);
  }

  onFontSelected(selectedFont) {
    this.executeCommand("change_text_font_overlay", [selectedFont]);
  }
  onTextColorChange(color) {
    this.executeCommand("change_text_color", [color]);
  }
  addStickerOverlay(imagePath) {
    this.executeCommand("add_image_sticker_overlay", [imagePath]);
  }

  addGifOverlay(gifPath) {
    this.executeCommand("add_gif_sticker_overlay", [gifPath]);
  }

  saveVideo() {
    this.executeCommand("save_video", []);
  }

  toggleShowCroppingView() {
    this.executeCommand("toggle_show_crop_preview", []);
  }

  onChangeTextBackground(background) {
    this.executeCommand("change_text_background_color", [background]);
  }

  onChangeTextAlignment(alignment) {
    this.executeCommand("change_text_alignment", [alignment]);
  }

  setAudioPath(audioPath) {
    this.executeCommand("audioPath", [audioPath]);
  }

  setVideoPath(videoPath) {
    this.executeCommand("videoPath", [videoPath]);
  }

  exportVideo() {
    this.executeCommand("exportVideo", []);
  }

  tagDraggable(tagArrayTemp) {
    this.executeCommand("tagUser", [tagArrayTemp]);
  }

  showTagTransView(showTransView){
    this.executeCommand("showTagTransparentView",[showTransView]);
  }

  setPreviousOverlayData(overlayDataList){
    this.executeCommand("restoreOverlay",overlayDataList);
  }

  playVideo(){
    this.executeCommand("playVideo",[]);
  }

  pauseVideo(){
    this.executeCommand("pauseVideo",[]);
  }

  getOverlayDataArray(callback){
    if(VideoEditorViewManager){
      VideoEditorViewManager.getOverlayDataArray(this._videoEditorHandle,callback);
    }
  }

  executeCommand(command, params) {
    UIManager.dispatchViewManagerCommand(
      this._videoEditorHandle,
      command,
      params,
    );
  }

  componentDidMount() {
    const eventEmitter = new NativeEventEmitter(VideoEditorViewManager);
    this.landscapeListener = eventEmitter.addListener(
      'EventLandscapeVideoCropping',
      (event) => {
        console.log(event);
        this.props.onCropping(event.xPosition);
      },
    );

    this.videoListener = eventEmitter.addListener(
      'EventGetVideoDetails',
      (videoDetails) => {
        console.log(videoDetails);
        this.props.videoDetails(videoDetails);
      },
    );

    this.videoSaveListener = eventEmitter.addListener(
      'EventOnVideoSaved',
      (videoDetails) => {
        console.log(videoDetails);
        this.props.onVideoSaved(videoDetails.videoPath);
      },
    );

    this.editedVideoPathListener = eventEmitter.addListener(
      'EventUpdateVideoPath',
      (updatedVideoDetails) => {
        this.props.updateVideoPath(updatedVideoDetails.videoPath);
      },
    );

    this.loaderVisibilityListener = eventEmitter.addListener(
      'EventShowOrHideLoader',
      (loaderMap) => {
        this.props.showOrHideLoader(loaderMap.showLoader);
      },
    );
    this.exportVideoListener = eventEmitter.addListener(
      'EventExportVideo',
      (videoDetails) => {
        this.props.onNext(videoDetails);
      },
    );

    this.eventTagPositionUpdate = eventEmitter.addListener(
      'EventUpdateClickedTagPosition',
      (tagPosUpdate) =>{
        this.props.tagUserPosClicked(tagPosUpdate);
      }
    );

    this.closeEditModeListener = eventEmitter.addListener(
      'EventCloseEditMode',
      () =>{
        this.props.closeEditMode();
      }
    );

    this.restoreOverlayListener = eventEmitter.addListener(
      'loadOverlayDataFromCache',
       () =>{
        this.props.restoreOverlay();
      }
    );
  }

  componentWillUnmount() {
    this.landscapeListener.remove();
    this.videoListener.remove();
    this.videoSaveListener.remove();
    this.editedVideoPathListener.remove();
    this.loaderVisibilityListener.remove();
    this.exportVideoListener.remove();
    this.eventTagPositionUpdate.remove();
    this.closeEditModeListener.remove();
    this.restoreOverlayListener.remove();
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <VideoEditorComponent ref={this._setReference} {...this.props} />
      </View>
    );
  }
}
