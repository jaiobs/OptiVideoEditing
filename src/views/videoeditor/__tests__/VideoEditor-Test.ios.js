import React from 'react';
import renderer from 'react-test-renderer';
import VideoEditor from '../VideoEditor.ios';
import {NativeModules} from 'react-native';

describe('VideoEditorTest', () => {
    beforeEach(() => {
        NativeModules.TextEmbedder = { 
            lockOrientationInPhotoView: (params) => {
                return Promise.resolve("locked");
            } 
        }
        NativeModules.PreviewFilter = { 
            updateVideoUrl: jest.fn()
        } 
      });

    it('should render correctly with state to props',()=> {
        const tree = renderer.create(
         <VideoEditor navigation={{ getParam: jest.fn()}}/>
      )
      .toJSON();
      expect(tree).toMatchSnapshot();
    })
  });