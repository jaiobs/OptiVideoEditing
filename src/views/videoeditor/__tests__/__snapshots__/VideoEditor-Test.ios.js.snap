// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoEditorTest should render correctly with state to props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <PreviewFilter
      filter={
        Object {
          "brightness": 0,
          "contrast": 1,
          "name": "",
          "range": 1,
          "saturation": 1,
          "type": "NORMAL",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": "100%",
          "position": "absolute",
          "width": "79%",
          "zIndex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 15,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "textShadowColor": "rgba(0, 0, 0, 0.75)",
            "textShadowOffset": Object {
              "height": 1,
              "width": -1,
            },
            "textShadowRadius": 10,
          }
        }
      >
        tap screen for filters
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 28,
        "height": 32,
        "justifyContent": "center",
        "marginLeft": 3,
        "position": "absolute",
        "right": 0,
        "width": 50,
        "zIndex": 999,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/res/assets/images/next_button_send.png",
          }
        }
        style={
          Array [
            Object {
              "height": 32,
              "margin": 10,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 13,
              "resizeMode": "contain",
              "width": 32,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": 50,
        "position": "absolute",
        "width": 50,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "left": 10,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "position": "absolute",
          "top": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/res/assets/images/cross.png",
          }
        }
        style={
          Object {
            "height": 24,
            "resizeMode": "contain",
            "width": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;
